### DISCUSSION TITLE
Help With GTK Source View Excel Macro language Spec File - for use with CherryTree Note Editor App

### DISCUSSION THREAD

**Discussion by timcrouse** [elementary/code#1221] [Q&A]
Could someone help me troubleshoot why this language spec file I generated does not invoke any syntax highlighting of any sort?

I created this file following the tutorial located here:
https://gnome.pages.gitlab.gnome.org/gtksourceview/gtksourceview5/lang-tutorial.html

Along with referring to the FSharp and Pascal language spec file and the def.lang file, INCLUDED IN THE ZIP FILE ALONG WITH VARIOUS EXCEL MACRO LAN SPEC FILES.

It is my understanding that def.lang is the top level file that defines styles that are allowed in language-specific spec files.

This language spec file is being tested in Cherry tree Portable, located here:
https://portableapps.com/apps/office/cherrytree-portable.

HERE ARE THE DESIRED COLOR GROUPS,
it is my understanding that the language spec file is how the app determines what to color and that the actual colors are applied via a selected style which is another file.  I presented the color with the groups in-case someone has experience creating a style definition that they could offer some guidance on that as well:

COMMENTS
 (all to be colored the same)  "GREEN"
REM  comment text xxxx
 //  comment text xxx
+N(  comment xxxxx )

Block Comment
//*  comment text xxxx
comment text xxxx  *//

The +N( xxx ) is the actual comment syntax allowed in an excel macro.  The other comment formats are desired  for commenting in
the raw text file outside of excel.

STRING FUNCTIONS  "MAGENTA"

ALL MATH FUNCTIONS  "BLUE"
MATH OPERATOR(S)  MOD, /, *, +, -, EXP,  SQRT  "DARK BLUE"
SPECIAL CHARACTER &,    "DARK BLUE"

INTEGER " BLUE,
FLOATING POINT "ORANGE"
BASE N  "CYAN"

BOOLEAN FUNCTIONS  "PURPLE"

LOOKUP & REF FUNCTIONS  "DARK RED"

[excelmacro.zip](https://github.com/elementary/code/files/10369700/excelmacro.zip)

**Comment by tintou** [comment]
You would have better answers on GNOME Discourse https://discourse.gnome.org/c/platform/5 which is the place to ask such questions to upstream developers
